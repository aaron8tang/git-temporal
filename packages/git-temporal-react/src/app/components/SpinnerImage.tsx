import React from 'react';

export interface SpinnerImageProps {
  // Style of the div the SVG is rendered in
  width?: number;
  height?: number;
}

export const SpinnerImage = (props: SpinnerImageProps): JSX.Element => {
  const { height = 200, width = 200 } = props;
  return (
    <svg
      height={height}
      width={width}
      version="1.1"
      id="L2"
      xmlns="http://www.w3.org/2000/svg"
      x="0px"
      y="0px"
      viewBox="0 0 100 100"
      enableBackground="new 0 0 100 100"
    >
      <circle
        fill="none"
        stroke="#ccc"
        strokeWidth="4"
        strokeMiterlimit="10"
        cx="50"
        cy="50"
        r="48"
      />
      <line
        fill="none"
        strokeLinecap="round"
        stroke="#ccc"
        strokeWidth="4"
        strokeMiterlimit="10"
        x1="50"
        y1="50"
        x2="85"
        y2="50.5"
      >
        <animateTransform
          attributeName="transform"
          dur="2s"
          type="rotate"
          to="0 50 50"
          from="360 50 50"
          repeatCount="indefinite"
        />
      </line>
      <line
        fill="none"
        strokeLinecap="round"
        stroke="#ccc"
        strokeWidth="4"
        strokeMiterlimit="10"
        x1="50"
        y1="50"
        x2="49.5"
        y2="74"
      >
        <animateTransform
          attributeName="transform"
          dur="15s"
          type="rotate"
          to="0 50 50"
          from="360 50 50"
          repeatCount="indefinite"
        />
      </line>
    </svg>
  );
};

SpinnerImage.displayName = 'SpinnerImage';
