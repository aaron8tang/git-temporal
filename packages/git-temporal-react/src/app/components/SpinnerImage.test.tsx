import React from 'react';
import { shallow } from 'enzyme';

import { SpinnerImage } from './SpinnerImage';

describe('components/SpinnerImage', () => {
  describe('when rendered with required props', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<SpinnerImage />);
    });

    test('it should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
