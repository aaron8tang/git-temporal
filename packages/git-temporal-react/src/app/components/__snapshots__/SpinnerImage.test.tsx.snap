// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SpinnerImage when rendered with required props it should match snapshot 1`] = `
<svg
  enableBackground="new 0 0 100 100"
  height={200}
  id="L2"
  version="1.1"
  viewBox="0 0 100 100"
  width={200}
  x="0px"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <circle
    cx="50"
    cy="50"
    fill="none"
    r="48"
    stroke="#ccc"
    strokeMiterlimit="10"
    strokeWidth="4"
  />
  <line
    fill="none"
    stroke="#ccc"
    strokeLinecap="round"
    strokeMiterlimit="10"
    strokeWidth="4"
    x1="50"
    x2="85"
    y1="50"
    y2="50.5"
  >
    <animateTransform
      attributeName="transform"
      dur="2s"
      from="360 50 50"
      repeatCount="indefinite"
      to="0 50 50"
      type="rotate"
    />
  </line>
  <line
    fill="none"
    stroke="#ccc"
    strokeLinecap="round"
    strokeMiterlimit="10"
    strokeWidth="4"
    x1="50"
    x2="49.5"
    y1="50"
    y2="74"
  >
    <animateTransform
      attributeName="transform"
      dur="15s"
      from="360 50 50"
      repeatCount="indefinite"
      to="0 50 50"
      type="rotate"
    />
  </line>
</svg>
`;
